src/types/Utf8Utf16Convert.cpp

```diff
@@ -208,161 +208,30 @@ void u16state::reset() noexcept
-                                    (gsl::narrow_cast<uint8_t>(*(it8 + 2)) ^ 0x00000080u);
-                    }
-                }
-
-                it8 += cnt;
-                if (cnt < 3u)
-                {
-                    hRes = S_FALSE;
-                }
-            }
-            // valid four bytes
-            // - F0     | 90..BF | 80..BF | 80..BF
-            // - F1..F3 | 80..BF | 80..BF | 80..BF
-            // - F4     | 80..8F | 80..BF | 80..BF
-            else if (gsl::narrow_cast<uint8_t>(*it8) >= 0xF0u && gsl::narrow_cast<uint8_t>(*it8) <= 0xF4u)
-            {
-                size_t cnt{ 1u };
-                if ((it8 + 1) < end8 &&
-                    (( // F0     | *90*..BF
-                         gsl::narrow_cast<uint8_t>(*it8) == 0xF0u &&
-                         gsl::narrow_cast<uint8_t>(*(it8 + 1)) >= 0x90u && gsl::narrow_cast<uint8_t>(*(it8 + 1)) <= contEnd) ||
-                     ( // F1..F3 | 80..BF
-                         gsl::narrow_cast<uint8_t>(*it8) >= 0xF1u && gsl::narrow_cast<uint8_t>(*it8) <= 0xF3u &&
-                         gsl::narrow_cast<uint8_t>(*(it8 + 1)) >= contBegin && gsl::narrow_cast<uint8_t>(*(it8 + 1)) <= contEnd) ||
-                     ( // F4     | 80..*8F*
-                         gsl::narrow_cast<uint8_t>(*it8) == 0xF4u &&
-                         gsl::narrow_cast<uint8_t>(*(it8 + 1)) >= contBegin && gsl::narrow_cast<uint8_t>(*(it8 + 1)) <= 0x8Fu)))
-                {
-                    ++cnt;
-                    if ((it8 + 2) < end8 && gsl::narrow_cast<uint8_t>(*(it8 + 2)) >= contBegin && gsl::narrow_cast<uint8_t>(*(it8 + 2)) <= contEnd)
-                    {
-                        ++cnt;
-                        if ((it8 + 3) < end8 && gsl::narrow_cast<uint8_t>(*(it8 + 3)) >= contBegin && gsl::narrow_cast<uint8_t>(*(it8 + 3)) <= contEnd)
-                        {
-                            ++cnt;
-                            codePoint = ((gsl::narrow_cast<uint8_t>(*it8) ^ 0x000000F0u) << 18u) |
-                                        ((gsl::narrow_cast<uint8_t>(*(it8 + 1)) ^ 0x00000080u) << 12u) |
-                                        ((gsl::narrow_cast<uint8_t>(*(it8 + 2)) ^ 0x00000080u) << 6u) |
-                                        (gsl::narrow_cast<uint8_t>(*(it8 + 3)) ^ 0x00000080u);
-                        }
-                    }
-                }
+        int lengthRequired{};
+        RETURN_HR_IF(E_ABORT, FAILED(SizeTToInt(in.length(), &lengthRequired)));
 
-                it8 += cnt;
-                if (cnt < 4u)
-                {
-                    hRes = S_FALSE;
-                }
-            }
-            else
-            {
-                hRes = S_FALSE;
-                ++it8;
-            }
+        out.resize(in.length()); // avoid to call MultiByteToWideChar twice only to get the required size
+        const int lengthOut = MultiByteToWideChar(65001u, 0ul, in.data(), lengthRequired, out.data(), lengthRequired);

```

Comment:
65001u I believe is defined globally as `CP_UTF8` by one of the Windows headers. If it's not available in this space, perhaps we should just redefine it at the top instead of having the magic value here.

Reply:
This was shorthand to avoid another narrow_cast since CP_UTF8 is an int.